pkg/migration/controllers/migration.go

```diff
@@ -1564,6 +1575,82 @@ func (m *MigrationController) getParsedLabels(
 	return m.getParsedMap(labels, clusterPair)
 }
 
+// updateStorageClassOnPV updates StorageClass on PV object
+// StorageClass is created on destination if it doesn't exist
+func (m *MigrationController) updateStorageClassOnPV(
+	migration *stork_api.Migration,
+	objects []runtime.Unstructured,
+	clusterPair *stork_api.ClusterPair,
+) error {
+	remoteClient, err := m.getRemoteClient(migration)
+	if err != nil {
+		return err
+	}
+	// Get a list of StorageClasses from destination
+	destStorageClasses, err := remoteClient.adminClient.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	destScExists := make(map[string]bool)
+	for _, sc := range destStorageClasses.Items {
+		destScExists[sc.Name] = true
+	}
+
+	for _, obj := range objects {
+		metadata, err := meta.Accessor(obj)
+		if err != nil {
+			return err
+		}
+		if obj.GetObjectKind().GroupVersionKind().Kind != "PersistentVolume" {
+			continue
+		}
+		destPV, err := remoteClient.adminClient.CoreV1().PersistentVolumes().Get(context.TODO(), metadata.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if scName, ok := destPV.Annotations[resourcecollector.CurrentStorageClassName]; ok {
+			// Create StorageClass on destination if it doesn't exist
+			if _, ok := destScExists[scName]; !ok {
+				// Get StorageClass from source
+				sc, err := storkcache.Instance().GetStorageClass(scName)
+				if err != nil {
+					return err
+				}
+				sc.UID = ""
+				sc.ResourceVersion = ""
+				log.MigrationLog(migration).Infof("Applying %v %v", sc.Kind, sc.Name)
+				for retries := 0; retries < maxApplyRetries; retries++ {
+					_, err = remoteClient.adminClient.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+					if err == nil || errors.IsAlreadyExists(err) {
```

Comment:
We may hit the nil pointer derereference. 
Lets make it like 
if err == nil || ( err != nil && errors.IsAlreadyExists(err))

Reply:
rhs of || will only be computed if err is not nil. So we don't need to check err != nil  condition explicitly.

Reply:
right. Ignore the comment.